gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/InMemoryGlobberFileSystem.java

```diff
@@ -105,7 +106,16 @@ public Path getWorkingDirectory() {
       throw new FileNotFoundException(
           String.format("Path '%s' (qualified: '%s') does not exist.", f, qualifiedPath));
     }
-    return fileStatuses.toArray(new FileStatus[0]);
+
+    FileStatus[] fileStatusArr = new FileStatus[fileStatuses.size()];
+    int i = 0;
+
+    for (FileStatus fs : fileStatuses)
+    {
+      fileStatusArr[i++] = new FileStatus(fs);
+    }
+
```

Comment:
Use regular for loop, because anyway you need an index:

```java
for (int i = 0; i < fileStatusArr.length; i++) {
  fileStatusArr[i] = new FileStatus(fileStatuses.get(i));
}
```